### PR TITLE
mcs: use release mode for gin

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5836.

### What is changed and how does it work?
This PR did some small improvements including using release mode for gin; omitting the resource manager for now; using revision for the watcher.

```commit-message
```

### Check List

Tests

- Unit test

```release-note
None.
```

### CODE REVIEW

**Review by binshi-bing — APPROVED** [review]
LGTM

File: server/server.go [hunk 1]
```diff
@@ -535,19 +534,13 @@ func (s *Server) LoopContext() context.Context {
...
-	if s.IsAPIServiceMode() { // disable tso service and resource manager service in api server
-		s.serverLoopWg.Add(2)
-		go s.watchServicePrimaryAddrLoop(mcs.TSOServiceName)
-		go s.watchServicePrimaryAddrLoop(mcs.ResourceManagerServiceName)
-	} else { // enable tso service
-		s.serverLoopWg.Add(1)
-		go s.tsoAllocatorLoop()
-	}
+	go s.tsoAllocatorLoop()
+	go s.watchServicePrimaryAddrLoop(mcs.TSOServiceName)
```
Comment (lhy1024) [inline server/server.go:543]: We will always run these goroutine whatever mode?

**Review by binshi-bing — APPROVED** [review]
LGTM

**Review by lhy1024 — APPROVED** [review]

File: server/server.go [hunk 2]
```diff
@@ -540,10 +541,9 @@ func (s *Server) startServerLoop(ctx context.Context) {
 	go s.etcdLeaderLoop()
 	go s.serverMetricsLoop()
 	go s.encryptionKeyManagerLoop()
-	if s.IsAPIServiceMode() { // disable tso service and resource manager service in api server
-		s.serverLoopWg.Add(2)
+	if s.IsAPIServiceMode() { // disable tso service in api server
+		s.serverLoopWg.Add(1)
 		go s.watchServicePrimaryAddrLoop(mcs.TSOServiceName)
-		go s.watchServicePrimaryAddrLoop(mcs.ResourceManagerServiceName)
```
Comment (lhy1024) [inline server/server.go:546]: Why remove it?

Comment (lhy1024) [reply server/server.go:546]: If remove it, the related test also need to be deleted or commented.

Comment (rleungx) [reply server/server.go:546]: We don't need it for now.

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**